components/automate-cli/cmd/chef-automate/pullAndGenerateConfig.go

```diff
@@ -681,21 +678,13 @@ func setExternalPGDetails(instanceUrl, superUserName, superUserPassword, dBUserN
 	}
 }
 
-func getChefServerFqdnAndLBRootCA(deploymentType string) (string, string, error) {
-	if deploymentType == EXISTING_INFRA_MODE {
-		ExistingInfraConfig, err := getExistingInfraConfig(filepath.Join(initConfigHabA2HAPathFlag.a2haDirPath, "config.toml"))
-		if err != nil {
-			return "", "", err
-		}
-		return ExistingInfraConfig.ChefServer.Config.RootCA, ExistingInfraConfig.ChefServer.Config.Fqdn, nil
-	} else if deploymentType == AWS_MODE {
-		AWSConfig, err := getAwsConfig(filepath.Join(initConfigHabA2HAPathFlag.a2haDirPath, "config.toml"))
-		if err != nil {
-			return "", "", err
+func getChefServerFqdnAndLBRootCA(config map[string]*dc.AutomateConfig) (string, string) {
+	for _, ele := range config {
+		if ele.Global != nil && ele.Global.V1 != nil && ele.Global.V1.ChefServer != nil && len(strings.TrimSpace(ele.Global.V1.ChefServer.Fqdn.Value)) > 0 && len(strings.TrimSpace(ele.Global.V1.ChefServer.RootCa.Value)) > 0 {
+			return ele.Global.V1.ChefServer.Fqdn.Value, ele.Global.V1.ChefServer.RootCa.Value
```

Comment:
make sure it is getting called in case of post deployment

Reply:
Yes this change is in pull and generate and it's getting called after deployment only.